### PR TITLE
Cli error cleanups

### LINKED ISSUE #761: We should give a better error if you run `hz serve` without a PEM file.
Currently we produce this output:

```
serve failed with Error: Could not access file "./horizon-key.pem" for running HTTPS server: Error: ENOENT: no such file or directory, open '/home/mlucy/hzdemo/horizon-key.pem'
```

It would be nice if we gave an error that:
- Explains more clearly what went wrong if you don't know what a PEM file is.
- Recommends running with `--dev` during development.
- Possibly tells you how generate a self-signed cert for testing?
- Possibly links to a page explaining what certificates are and how to generate a self-signed one for testing?

### PR DESCRIPTION
**Pull request by deontologician**
Fixes #761 

As a part of it, I added a new `NiceError` class that formats errors better, and modified the main cli catch clause to use it.

Also caught a bug where some command flag options were being ignored.

The original goal of #761 was to give a better error when doing `hz serve` instead of `hz serve --dev`, since the exception you'll get is something about `horizon-cert.pem` and is confusing if you just want to get started. Now, if you do `hz serve`, you'll get:

selection_090

If you actually called `hz serve --cert-file=some_filename.pem`, it gives a more terse error:

selection_089

Assigning to @mlucy since he was the one who originally complained :P

---

This change is [Reviewable](https://reviewable.io/reviews/rethinkdb/horizon/776)

### CODE REVIEW

**Comment by deontologician** [comment]
Just noticed a stray `can to` in the suggestion for disabling security.

File: cli/src/serve.js [hunk 1]
```diff
@@ -244,18 +245,50 @@ const create_insecure_servers = (opts) => {
...
+them in your browser.`
+      );
+      description = `In order to run the server in secure mode (the default), \
+Horizon needs both a certificate file and a key file to encrypt websockets. \
+By default, it looks for horizon-key.pem and horizon-cert.pem \
+files in the current directory.`;
+    } else {
+      // They supplied a cert or key file, so don't give the long
+      // explanation and irrelevant suggestions.
+      suggestions.unshift(`See if the ${type} filename was misspelled`);
```
Comment (mlucy) [inline cli/src/serve.js]: I think this should have a `.` at the end (all the other list items do).

File: cli/src/utils/nice_error.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        results.push.apply(results, formattedSrc);
+      }
+    }
+    if (this.suggestions) {
+      results.push(
+        '', // extra newline before suggestions
+        chalk.red(
+          this.suggestions.length > 1 ? 'Suggestions:' : 'Suggestion:'));
+      results.push.apply(
+        results, this.suggestions.map((note) => `  ➤ ${note}`));
```
Comment (mlucy) [inline cli/src/utils/nice_error.js:51]: How strongly attached are you to using a Unicode character as the list element prefix?  It struck me as visually weird compared to `-` or `*`, which I'm used to seeing for lists in error messages, and it won't always display correctly for people.

**Comment by mlucy** [comment]
Looks good to me aside from the two minor comments.

File: cli/src/utils/nice_error.js (hunk 2 above)
Comment (deontologician) [reply cli/src/utils/nice_error.js:51]: We use unicode elsewhere, and I tried using *, the problem was with very long suggestions it wasn't visually obvious where one started and the next began

**Pull request closed without merging**